### ISSUE TITLE
TBB and performance.

### ISSUE THREAD

**Issue by YooSunYoung** [scipp/scipp#3551]
We had an issue with performance processing histogram mode detector data for ymir demo.
It was because we didn't have TBB library installed.

TBB library is available in `conda` but not in `pypi`,
which means when `scipp` is insatlled via `conda`, it was faster.

We need to find the way to install TBB even if we use `pypi` to install `scipp`.

It may vary for platforms, `windows`, `MacOS` and `Linux`.

TODO
 - [ ] Find the easiest/best way to install TBB with scipp
 - [ ] Test installing them in VISA (on different platforms)
 - [ ] Update scipp isntallation guide in the docs
 - [ ] Discuss with DST about how to make sure it is deployed.

**Comment by SimonHeybrock** [comment]
I am surprised that Scipp would even work without TBB. Are you sure about it missing completely?

**Comment by jl-wynen** [comment]
I thought TBB is bundled with the wheel. Doesn't cibuildwheel do this?

**Comment by SimonHeybrock** [comment]
Yes it does, you are right. So maybe the only missing bit is the extra lib for the allocation proxy?

**Comment by MridulS** [comment]
Someone pointed me towards this https://scipp.github.io/user-guide/tips-tricks-and-anti-patterns.html#Using-the-TBB-allocation-proxy once.

**Comment by SimonHeybrock** [comment]
@MridulS the point here is that this is not available since there is an apparently optional extra lib as part of TBB, which is not installed with `pip`.

**Comment by MridulS** [comment]
Just in case this is useful we do use different TBBs. In the pypi wheel it comes from conan (https://github.com/scipp/scipp/blob/2bc44c832daf1a1362915e02ae740bc8655dda51/lib/cmake/scipp-conan.cmake#L31) and the conda one packages tbb and tbb-devel

**Comment by SimonHeybrock** [comment]
> Just in case this is useful we do use different TBBs. In the pypi wheel it comes from conan (
> 
> https://github.com/scipp/scipp/blob/2bc44c832daf1a1362915e02ae740bc8655dda51/lib/cmake/scipp-conan.cmake#L31
> ) and the conda one packages tbb and tbb-devel

Yes, I think this is the entire point: When building the wheel, apparently not all libs are included, whereas the conda package has it.

**Comment by jokasimr** [comment]
@YooSunYoung how big was the difference?

**Comment by YooSunYoung** [comment]
> how big was the difference?

With the particular workflow that we were running, it was from ~10 minutes to ~10 seconds.

**Comment by YooSunYoung** [comment]
>There is https://pypi.org/project/tbb, what happens if we just install it in the env?

@MridulS Hmm I haven't tried. Can you please try it on VISA...? It's most relevant there.
